File: leetcode/0370-range-addition/solution.py
from typing import List


class Solution:
    def getModifiedArray(self, length: int, updates: List[List[int]]) -> List[int]:
        diff = [0] * (length + 1)
        for start, end, inc in updates:
            diff[start] += inc
            diff[end + 1] -= inc

        result = [0] * length
        running = 0
        for i in range(length):
            running += diff[i]
            result[i] = running
        return result

File: leetcode/0370-range-addition/solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> getModifiedArray(int length, vector<vector<int>>& updates) {
        vector<int> diff(length + 1, 0);
        for (auto& update : updates) {
            int start = update[0];
            int end = update[1];
            int inc = update[2];
            diff[start] += inc;
            diff[end + 1] -= inc;
        }

        vector<int> result(length, 0);
        int running = 0;
        for (int i = 0; i < length; i++) {
            running += diff[i];
            result[i] = running;
        }
        return result;
    }
};
